data/rulesets.ts

```diff
@@ -2965,4 +2965,175 @@ export const Rulesets: import('../sim/dex-formats').FormatDataTable = {
 			return newSpecies;
 		},
 	},
+	datapreview: {
+		effectType: 'Rule',
+		name: 'Data Preview',
+		desc: 'When a new Pokémon switches in for the first time, information about its types, stats and Abilities is displayed to both players.',
+		onSwitchIn(pokemon) {
+			let species = this.dex.species.get(pokemon.species.name);
+			const gen = this.gen;
+
+			if (pokemon.illusion) {
+				species = this.dex.species.get(pokemon.illusion.species.name);
```

Comment:
Does this all need to be done twice? Can you just do

```ts
			let species = this.dex.species.get(pokemon.species.name);
			const gen = this.gen;

			if (pokemon.illusion) {
				species = this.dex.species.get(pokemon.illusion.species.name);
			}

			// Recreation of Chat.getDataPokemonHTML
```

And then go from there?

Reply:
Great catch, as long as I replace species and the boolean that the real species hasn't been revealed, that works fine. It's now been modified.